bugbear.py

```diff
@@ -67,8 +67,31 @@ def gen_line_based_checks(self):
         The following simple checks are based on the raw lines, not the AST.
         """
         for lineno, line in enumerate(self.lines, start=1):
+            # Special case: ignore long shebang (following pycodestyle).
+            if lineno == 1 and line.startswith("#!"):
+                continue
+
             length = len(line) - 1
             if length > 1.1 * self.max_line_length:
+                # Special case long URLS and paths to follow pycodestyle.
+                # Would use the `pycodestyle.maximum_line_length` directly but
+                # need to supply it arguments that are not available so chose
+                # to replicate instead.
+                chunks = line.split()
+
+                is_line_comment_url_path = len(chunks) == 2 and chunks[0] == "#"
+
+                just_long_url_path = len(chunks) == 1
```

Comment:
So if I read this right, really this is just a long no white space string right? So it could not be a URL or file path? I guess generally we'll have spaces I guess is the rationale here right?

Is this the same as how pycodestyle does it? if so, (Sorry - being lazy and not checking), then I guess it's proven to be safe enough.

Reply:
Yeah, a long string with no white space, which is mostly just URLs and paths. A path that has a space in it will still fail but would also fail E501. Rational is it is more readable to not split up URLs and paths in comments and code.

It's the same logic as pycodestyle, modified to be a bit more readable to me. Linking [here](https://github.com/PyCQA/pycodestyle/blob/main/pycodestyle.py#L267-L296) for convenience.